app/models/mixins/class_from_string_mixin.rb

```diff
@@ -0,0 +1,12 @@
+module ClassFromStringMixin
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def class_from_string(type)
+      return self if type == to_s || type.nil?
```

Comment:
@Fryguy wasn't sure we should default to returning itself if type is nil. We did this in Authentication, not sure we should generalize that though. thoughts?